pandas/core/indexes/base.py

```diff
@@ -4790,7 +4791,7 @@ def groupby(self, values) -> Dict[Hashable, np.ndarray]:
         # map to the label
         result = {k: self.take(v) for k, v in result.items()}
 
-        return result
+        return PrettyDict(result)
```

Comment:
could also update return type on L4769 and docstring. would need PrettyDict to be generic, see other comment.